File: BlazorPaintComponent/BPaintCJsInterop.cs
using Microsoft.JSInterop;
using System.Threading.Tasks;

namespace BlazorPaintComponent
{
    public class BPaintCJsInterop
    {
        public static IJSRuntime jsRuntime;

        public static ValueTask<string> alert(string message)
        {

            return jsRuntime.InvokeAsync<string>(
                "BPaintCJsInterop.alert",
                message);
        }


        public static ValueTask<string> log(string message)
        {

            return jsRuntime.InvokeAsync<string>(
                "BPaintCJsInterop.log",
                message);
        }

        public static ValueTask<bool> GetElementBoundingClientRect(string id, DotNetObjectReference<CompBlazorPaint> dotnethelper)
        {

            return jsRuntime.InvokeAsync<bool>(
                "BPaintCJsInterop.GetElementBoundingClientRect",
                new { id, dotnethelper });
        }

        public static ValueTask<bool> UpdateSVGPosition(string id)
        {
     
            return jsRuntime.InvokeAsync<bool>(
                "BPaintCJsInterop.UpdateSVGPosition", id);
        }


        public static ValueTask<bool> SetCursor(string cursorStyle = "default")
        {

            return jsRuntime.InvokeAsync<bool>(
                "BPaintCJsInterop.SetCursor",
                cursorStyle);
        }
    }
}

File: BlazorPaintComponent/wwwroot/BPaintCJsInterop.js
window.BPaintCJsInterop = {
    alert: function (message) {
        return alert(message);
    },
    log: function (message) {
        return log(message);
    },
    GetElementBoundingClientRect: function (obj) {

        if (document.getElementById(obj["id"]) !== null) {
            let rect = document.getElementById(obj["id"]).getBoundingClientRect();

            let myleft = rect.left.toFixed(2) + window.scrollX;
            let mytop = rect.top.toFixed(2) + window.scrollY;

            obj["dotnethelper"].invokeMethodAsync('invokeFromjs', obj["id"], myleft, mytop);
            return true;
        }
        else {
            return false;
        }
    },
    UpdateSVGPosition: function (id) {
            if (document.getElementById(id) !== null) {
               
                let rect = document.getElementById(id).getBoundingClientRect();

                let myleft = rect.left.toFixed(2) + window.scrollX;
                let mytop = rect.top.toFixed(2) + window.scrollY;

                DotNet.invokeMethodAsync('BlazorPaintComponent', 'invokeFromjs_UpdateSVGPosition', myleft, mytop);
               
                return true;
            }
            else {
                return false;
            }
       
    },
    SetCursor: function (cursorStyle) {
        document.body.style.cursor = cursorStyle;
        return true;
    },
};